### ISSUE TITLE
[HOLD for payment 2023-12-07] [$125]  IOU Receipt - Attachment thumbnail appears when returning online after removing receipt offline

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#29462]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.83-1

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com
2. Go offline
3. Go to + > Request money > Scan > Upload a receipt > Create the request
4. Open the Scan request in details page
5. Manually enter the amount and merchant
6. Click on the receipt > three-dot menu > Delete receipt > Delete
7. Return online
Note that after a while, the attachment thumbnail returns and it loads infinitely when clicking on it

## Expected Result:
There will be no receipt in the money request since the receipt is removed

## Actual Result:
Attachment thumbnail appears in the money request a while after going online. When tapping on it, it loads infinitely

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

 

https://github.com/Expensify/App/assets/78819774/6699cac0-1b17-4d96-b54a-252a068e68d6

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0190ee90fc1a86933e</li>
        <li>Upwork Job ID: 1712522397261840384</li>
        <li>Last Price Increase: 2023-11-08</li>
    </ul>
</details>

**Comment by mountiny** [comment]
cc @luacmartins

**Comment by jasperhuangg** [comment]
@lanitochka17 I don't see a "Delete receipt" button anywhere in the App, and the screen recording doesn't align with the reproduction steps. Can you confirm?

**Comment by jasperhuangg** [comment]
Reproduction steps aren't clear, I'm going to remove the `Deploy Blocker` label for now.

**Comment by jasperhuangg** [comment]
Adding the label back, the instructions were clarified [here](https://expensify.slack.com/archives/C9YU7BX5M/p1697148532019349)

**Comment by jasperhuangg** [comment]
I don't think we need to be blocking deploy on this, in an actual use case if the user is offline the entire time then there's no way their UI would update to include the "Delete Receipt" option in the 3 dots menu.

**Comment by luacmartins** [comment]
I can work on this

**Comment by luacmartins** [comment]
Have a draft PR up, but I still see the issue. I'll have to investigate this further

**Comment by Christinadobrzyn** [comment]
Should this be internal? @luacmartins @aimane-chnaif?

**Comment by luacmartins** [comment]
Yes, updated the issue. Thanks!

**Comment by luacmartins** [comment]
Same as above. I'll try to work on this next week.

**Comment by luacmartins** [comment]
I'll work on it this week.

**Comment by Christinadobrzyn** [comment]
Thanks @luacmartins!

**Comment by luacmartins** [comment]
This is an internal issue, but not a priority. I'll get to it soon

**Comment by luacmartins** [comment]
Still not a priority.

**Comment by luacmartins** [comment]
I looked at this yesterday and it doesn't seem to be a BE issue. Even when we do return the onyxData from `DetachReceipt` we somehow still end up with incorrect data in Onyx. Gonna make this external again.

**Comment by esh-g** [comment]
@luacmartins Actually I think this is BE bug because the BE sends a `mergeCollection` command for `transaction` with receipt but since we are setting the receipt to `{}` we should do a `SET` instead, just like we do in the `detachReceipt` function optimistic data.

**Comment by luacmartins** [comment]
Which request is sending the `mergeCollection`?

**Comment by esh-g** [comment]
If you're wondering why the optimistic data from the `detachReceipt` function doesn't persist, it is because with the `ReconnectApp` request, the receipt is set once again and the optimistic removal gets undone.

<img width="547" alt="Screenshot 2023-11-08 at 4 49 17 PM" src="https://github.com/Expensify/App/assets/77237602/be6d4b8d-3cd6-4fde-9956-3d9a78ac3b81">

**Comment by luacmartins** [comment]
@esh-g I see your point, but even when I update the API to return a set to null so that's the last update in ReconnectApp, the issue is still not resolved:

![Screenshot 2023-11-08 at 2 15 51 PM](https://github.com/Expensify/App/assets/22219519/5168694f-bc93-48f5-989d-ad6ab7cd2807)
![Screenshot 2023-11-08 at 2 16 11 PM](https://github.com/Expensify/App/assets/22219519/e2e9d568-70cb-480a-95a1-d6df918ba698)

**Comment by esh-g** [comment]
hmm @luacmartins that seems a little strange...
Are you sure that the receipt is not added back to the transaction in the `ReconnectToReport` calls after the `ReconnectApp` call?

Because I don't think there is any front-end code that changes the transaction data in onyx, because optimistic data is set correctly and there is no `successData` passed so, the change to the transaction must come from the backend when the `DetachReceipt` request succeeds...

**Comment by luacmartins** [comment]
I see the calls to `ReconnectToReport` but they all have `receipt: {}`, which when merged after the `null` value I showed above wouldn't bring the receipt back. I agree with you that the issue happens once we get the API response back though. Maybe the onyx merge/set is not happening in the correct order or there's another update object that I'm missing.

**Comment by Christinadobrzyn** [comment]
just checking, we're still working on a proposal, correct? 

This was automatically changed to an internal issue due to it being over 4 weeks. @luacmartins should this be switched to Weekly and external again?

**Comment by Christinadobrzyn** [comment]
I'm going ooo until Monday Nov 27th so going to assign another BZ teammate to monitor this until I'm back. Thanks!

Status: waiting on an update from @luacmartins when possible!

**Comment by luacmartins** [comment]
We still haven't found the root cause here. I'll put a PR to start sending the correct Onyx update from the API, but that hasn't solved the issue for me locally. We can investigate further once the API changes are live.

**Comment by Christinadobrzyn** [comment]
back from ooo - taking this back from Anu. Looks like we're working on the [PR](https://github.com/Expensify/App/pull/31850)

**Comment by Christinadobrzyn** [comment]
Prepping for payment

Payouts due:

Issue Reporter: NA
Contributor: NA
Contributor+: $125 @aimane-chnaif (in Upwork)

Eligible for 50% #urgency bonus? N

Upwork job is [here](https://www.upwork.com/jobs/~0142c298d4bd029a0d).

@aimane-chnaif do we need a regression test?

**Comment by Christinadobrzyn** [comment]
Awesome! Paid you in Upwork based on this payment structure https://github.com/Expensify/App/issues/29462#issuecomment-1841937946
